### PR TITLE
refactor(module:date-picker): add new type for date-picker size

### PR DESCRIPTION
**Pull request by DepickereSven**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
There wasn't a type being used to define the possible sizes, like in other places.
Example reference: https://github.com/NG-ZORRO/ng-zorro-antd/blob/b215efa41a78289c07efb66b006781649de9a74d/components/select/select.component.ts#L56
Issue Number: N/A

## What is the new behavior?
Created a type in the example of `NzSelectSizeType `

## Does this PR introduce a breaking change?
```
[x] Yes
[] No
```

You can now use the type `NzDatePickerType` to define your `nzSize`.

### CODE REVIEW

File: components/date-picker/date-picker.component.ts [hunk 1]
```diff
@@ -41,6 +41,8 @@ import { CompatibleDate, DisabledTimeFn, NzDateMode, PresetRanges, SupportTimeOp
 const POPUP_STYLE_PATCH = { position: 'relative' }; // Aim to override antd's style to support overlay's position strategy (position:absolute will cause it not working beacuse the overlay can't get the height/width of it's content)
 const NZ_CONFIG_MODULE_NAME: NzConfigKey = 'datePicker';
 
+export type NzDatePickerType = 'large' | 'default' | 'small';
```
Comment (hsuanxyz) [inline components/date-picker/date-picker.component.ts:44]: Thank you for the PR, this type need to export in `date-picker/public-api.ts`

**Review by hsuanxyz — CHANGES_REQUESTED** [review]

File: components/date-picker/date-picker.component.ts (hunk 1 above)
Comment (wenqi73) [reply components/date-picker/date-picker.component.ts:44]: Use `NzDatePickerSizeType` would be better.

File: components/date-picker/public-api.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export * from './util';
 
 export { NzDatePickerModule } from './date-picker.module';
 export { NzDatePickerComponent } from './date-picker.component';
+export { NzDatePickerSizeType } from './date-picker.component';
```
Comment (hsuanxyz) [inline components/date-picker/public-api.ts:11]:
```suggestion
export { NzDatePickerComponent, NzDatePickerSizeType } from './date-picker.component';
```

**Review by hsuanxyz — CHANGES_REQUESTED** [review]

**Review by wenqi73 — APPROVED** [review]
LGTM

**Pull request merged**